rust/nix-expr/src/eval_state.rs

```diff
@@ -191,12 +237,24 @@ impl Drop for EvalState {
 #[cfg(test)]
 mod tests {
     use ctor::ctor;
+    use nix_util::settings;
 
     use super::*;
 
     #[ctor]
     fn setup() {
-        init().unwrap();
+        (|| -> Result<()> {
+            init()?;
+            // If it reinvokes the test suite,
+            // settings::set("build-hook", "")?;
+
+            // When testing in the sandbox, the default build dir would be a parent of the storeDir,
+            // which causes an error. So we set a custom build dir here.
+            settings::set("sandbox-build-dir", "/custom-build-dir-for-test")?;
+            Ok(())
+        })()
+        .unwrap();
```

Comment:
Making a lambda and then immediately calling is confusing to me. Is there a reason not to call `.unwrap()` on each of the lines that might fail?

Reply:
Done.

> refact: Distribute unwrap()
>
> This gives us better error messages in case it does go wrong.